### PR TITLE
classroomFinder-mc

### PR DESCRIPTION
**Pull request by mcpenguin**
- Separated client (front-end) and server (back-end)
- Set up backend (mongoose + express)
- Made simple route to fetch all classes in a particular building.

```
cd server
npm run build
npm start
```
Then go to `localhost:3000/classes/:buildingCode` (eg buildingCode=MC).

### CODE REVIEW

**Comment by mcpenguin** [comment]
Main updates:

I've updated the Class object (in the database). It now has the following form:
image

The main change is that `startTime` and `endTime` are now objects that hold hour and minute fields, which correspond to 24-hour times.

There are now two routes:
`/classes/:buildingCode?query=D` - returns all classes in building code (eg E2) that have a class on D, where
```
D = M | T | W | Th | F | S | Su
```
(which correspond to the respective days of the week).

`/classes/:buildingCode/:roomNumber?query=D` - returns all classes in building code (eg E2) and room number (eg 1736) that have a class on D, where D is as above.

File: client/package.json [hunk 1]
```diff
@@ -14,8 +14,12 @@
     "@react-navigation/bottom-tabs": "^6.2.0",
     "@react-navigation/native": "^6.0.8",
     "@react-navigation/stack": "^6.1.1",
+    "concurrently": "^7.0.0",
+    "cors": "^2.8.5",
     "expo": "~44.0.0",
     "expo-status-bar": "~1.2.0",
+    "express": "^4.17.3",
```
Comment (victorjzsun) [inline client/package.json]: Why does client need server-related packages?

File: .gitignore [hunk 2]
```diff
@@ -9,6 +9,9 @@ npm-debug.*
 *.mobileprovision
 *.orig.*
 web-build/
+.idea
+
+server/*.js
```
Comment (victorjzsun) [inline .gitignore]: This isn't needed, tsc puts everything in dist folder which is ignored. There might be some js files we need to keep in the future

**Comment by victorjzsun** [comment]
https://github.com/bonniepeng2002/WatIamLOST/pull/4#issuecomment-1072031079 can we put this in readme?

File: client/package.json [hunk 3]
```diff
@@ -14,6 +14,8 @@
     "@react-navigation/bottom-tabs": "^6.2.0",
     "@react-navigation/native": "^6.0.8",
     "@react-navigation/stack": "^6.1.1",
+    "concurrently": "^7.0.0",
+    "cors": "^2.8.5",
```
Comment (victorjzsun) [inline client/package.json]: All of these server-related packages can be removed

**Review by victorjzsun — APPROVED** [review]
lgtm

**Pull request merged**